### PR TITLE
docs:#1240 updated Tiered Storage required properties

### LINKED ISSUE #1240: update Tiered Storage required properties
### Describe the Issue

see https://redpandadata.slack.com/archives/C01LK3UK2K1/p1676468142450519

Our [tiered storage doc](https://docs.redpanda.com/docs/manage/tiered-storage/) is not correct wrt "required" properties:
Those properties are not required (and in fact should not be provided at all) if the recommended IAM role approach is taken to configuration (the approach mentioned near the start of the doc). In that case the credentials will come from the cloud source. 

UPDATE TO: 
cloud_storage_access_key
AWS or GCS access key.
Required for AWS and GCS authentication with access keys.

cloud_storage_secret_key 
AWS or GCS secret key.
Required for AWS and GCS authentication with access keys.

[@Abhijat Malviya](https://redpandadata.slack.com/team/U03BLA5EX2A) confirmed: This is correct, we had a bug in validation where the system still required the keys to be present even if IAM roles were used to auth (https://github.com/redpanda-data/redpanda/issues/6795) but that was fixed several months ago. When using IAM roles in redpanda, shouldn't set secret keys at the same time.

SME = Travis Downs

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/tiered-storage/

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1240 

see https://redpandadata.slack.com/archives/C01LK3UK2K1/p1676468142450519
this updates:
`cloud_storage_access_key`
AWS or GCS access key.
Required for AWS and GCS authentication with access keys.

`cloud_storage_secret_key`
AWS or GCS secret key.
Required for AWS and GCS authentication with access keys.

preview beta: https://deploy-preview-1266--redpanda-documentation.netlify.app/docs/beta/manage/tiered-storage/#tiered-storage-configuration-properties

preview 22.3: https://deploy-preview-1266--redpanda-documentation.netlify.app/docs/manage/tiered-storage/#tiered-storage-configuration-properties

### CODE REVIEW

File: docs/manage/tiered-storage.mdx [hunk 1]
```diff
@@ -409,9 +409,9 @@ The following list contains cluster-level configuration properties for Tiered St
 | `cloud_storage_enabled` | Global flag that enables Tiered Storage. <br/>Set to true to enable Tiered Storage. Default is false.  |
 | `cloud_storage_region` | Cloud storage region. <br/>Required for AWS and GCS. |
 | `cloud_storage_bucket` | AWS or GCS bucket name. <br/>Required for AWS and GCS. |
-| `cloud_storage_credentials_source` | AWS or GCS instance metadata. <br/>Required for AWS and GCS with IAM roles. |
-| `cloud_storage_access_key` | AWS or GCS access key. <br/>Required for AWS and GCS with access keys. |
-| `cloud_storage_secret_key` | AWS or GCS secret key. <br/>Required for AWS and GCS with access keys.  |
+| `cloud_storage_credentials_source` | AWS or GCS instance metadata. <br/>Required for AWS and GCS authentication with IAM roles. |
```
Comment (abhijat) [inline docs/manage/tiered-storage.mdx:412]: should be `AWS or GCP`?

Comment (Feediver1) [reply docs/manage/tiered-storage.mdx:412]: +1. Need to change to GCP @micheleRP

Comment (Feediver1) [reply docs/manage/tiered-storage.mdx:412]: ^^ Please make sure this is correct @micheleRP

**Review by Feediver1 — APPROVED** [review]
After confirming "GCS" or "GCP", lgtm

**Review by travisdowns — APPROVED** [review]
LGTM.

File: docs/manage/tiered-storage.mdx (hunk 1 above)
Comment (micheleRP) [reply docs/manage/tiered-storage.mdx:412]: @abhijat, @Feediver1: Per @mattschumpert, we should be using GCS.

**Pull request merged**